drgn_tools/cpuinfo.py

```diff
@@ -0,0 +1,581 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helper to view cpuinfo data
+"""
+import argparse
+from typing import Any
+from typing import Dict
+
+from drgn import Object
+from drgn import Program
+from drgn import sizeof
+from drgn.helpers.linux.bitops import for_each_set_bit
+from drgn.helpers.linux.bitops import test_bit
+from drgn.helpers.linux.percpu import per_cpu
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.table import print_dictionary
+
+
+# Constants
+X86_FEATURE_PTI = 7 * 32 + 11
+X86_FEATURE_IBPB = 7 * 32 + 26
+X86_FEATURE_USE_IBRS_FW = 7 * 32 + 22
+X86_FEATURE_RSB_CTXSW = 7 * 32 + 19
+X86_FEATURE_HYPERVISOR = 4 * 32 + 31
+X86_FEATURE_MSR_IA32_FEAT_CTL = 7 * 32 + 31
+X86_FEATURE_VMX = 4 * 32 + 5
+
+X86_BUG_MSBDS_ONLY = 20
+X86_BUG_MMIO_UNKNOWN = 26
+
+X86_CR4_VMXE = 1 << 13
+
+X86_VENDOR_HYGON = 2
+X86_VENDOR_AMD = 3
+
+SPEC_CTRL_IBRS_FIRMWARE = 1 << 3
+SPEC_CTRL_IBPB_INUSE = 1 << 0
+
+TAINT_NO_RETPOLINE = 16
```

Comment:
I searched high and low for where you possibly could have come from with this. It appears in [this](https://lore.kernel.org/all/20180112175507.31750-1-andi@firstfloor.org/) patch series but in no upstream kernel, ever.

It turns out that the series there got included into UEK4 for some reason...

At the point of the code where you use this, please do write a nice comment explaining what this is, which kernel versions it's valid for, and why it exists. As it is, taint #16 is `TAINT_AUX` in any normal kernel, so this is highly unusual.